dist-git.spec

```diff
@@ -164,15 +132,22 @@ cd -
 
 /usr/sbin/hardlink -cv %{buildroot}%{_datadir}/selinux
 
+%pre selinux
+if /usr/sbin/selinuxenabled ; then
+   [ -f %{file_context_file_pre} ] || cp -f %{file_context_file} %{file_context_file_pre}
+fi
 
 %post selinux
 for selinuxvariant in %{selinux_variants}
 do
   /usr/sbin/semodule -s ${selinuxvariant} -i \
     %{_datadir}/selinux/${selinuxvariant}/%{modulename}.pp &> /dev/null || :
 done
-%{_sbindir}/restorecon -Rv %{_sharedstatedir}/dist-git || :
 
+%posttrans selinux
+if /usr/sbin/selinuxenabled ; then
+    %{_sbindir}/restorecon -Rv /srv/web/ || :
+fi
```

Comment:
I didn't apply this, in the end. If %{file_context_file_pre} does not exist, whole filesystem is relabeled at installation of the dist-git-selinux package. The best way is probably to call restorecon only on /srv/web to avoid huge restorecons that does not really change the context. However restorecons on /srv/git and /srv/cache  should maybe be applied without -R.